### ISSUE TITLE
install.ps1 errors when installing Nancy.Viewengines.Razor.1.2.0 via Nuget

### ISSUE THREAD

**Issue by franhoey** [NancyFx/Nancy#2050]
When I update or install Nancy.Viewengines.Razor.1.2.0 via nuget I get the following errors from install.ps1:

> Unable to find type [NuGet.VisualStudio.IFileSystemProvider].
> At C:\Projects\Foremans\Saturn\Web\packages\Nancy.Viewengines.Razor.1.2.0\tools\install.ps1:15 char:5
> +     $fileSystemProvider = $componentModel.GetService([NuGet.VisualStu ...
> +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (NuGet.VisualStudio.IFileSystemProvider:TypeName) > > [], RuntimeException
>     + FullyQualifiedErrorId : TypeNotFound
>  
> Unable to find type [NuGet.VisualStudio.ISolutionManager].
> At C:\Projects\Foremans\Saturn\Web\packages\Nancy.Viewengines.Razor.1.2.0\tools\install.ps1:16 char:5
> +     $solutionManager = $componentModel.GetService([NuGet.VisualStudio ...
> +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (NuGet.VisualStudio.ISolutionManager:TypeName) [], RuntimeException
>     + FullyQualifiedErrorId : TypeNotFound
>  
> You cannot call a method on a null-valued expression.
> At C:\Projects\Foremans\Saturn\Web\packages\Nancy.Viewengines.Razor.1.2.0\tools\install.ps1:18 char:5
> +     $fileSystem = $fileSystemProvider.GetFileSystem($solutionManager. ...
> +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : InvokeMethodOnNull

I'm on a fresh install of Windows 10 and Visual Studio 2015

**Comment by khellang** [comment]
Hmm. This package seems to know a bit too much about NuGet, which has now changed significantly with V3 (shipped in Visual Studio 2015). We'll have to think about this...

**Comment by yishaigalatzer** [comment]
As @khellang is saying, this package is relying on internal implementation details of NuGet 2.X, which is why it fails running `install.ps1` with NuGet3.

Short term - NuGet for web projects still uses packages.config across all web projects. 
Longer term - We are tracking adding items like content and scaffolding (including transforms) to project.json scenarios, to make the web scenario applicable to project.json scenarios.

Post build step - I would try to avoid it, and use targets if possible. Because you want it to work outside of visual studio as well.

In general we advise against install.ps1 use, this issue is a classic example of why. Install.ps1 relies many times on implementation details and on running inside visual studio. Which makes it break version to version, and not applicable from command line and cross platform.

**Comment by thecodejunkie** [comment]
The problem is that we need to put a post-build step into the project so that we can copy a bundled dll file to the bin folder. This is not something that we invented, but it's the rather stupid design of Razor design time support.

**Comment by khellang** [comment]
@thecodejunkie This can be done like this;

https://github.com/khellang/ResourceHelperGenerator/blob/b14c638c956f902a9fa4a744d1b9139a0ca1d210/targets/ResourceHelperGenerator.targets
https://github.com/khellang/ResourceHelperGenerator/blob/b14c638c956f902a9fa4a744d1b9139a0ca1d210/ResourceHelperGenerator.nuspec#L19

Except this is a pre-build hook. It should be easy to do with targets.

**Comment by divmgl** [comment]
I'll have to skip on using this because I'm having the same issue as well.

```
Unable to find type [NuGet.VisualStudio.IFileSystemProvider]. Make sure that the assembly that contains this type is loaded.
```
```
Unable to find type [NuGet.VisualStudio.ISolutionManager]. Make sure that the assembly that contains this type is loaded.
```
Exception setting "Value": "Exception occurred. (Exception from HRESULT: 0x80020009 (DISP_E_EXCEPTION))"
```

**Comment by khellang** [comment]
Closed by #2101
